### PR TITLE
DPL-126 Biosero plate state

### PR DESCRIPTION
**Pull request by BenTopping**
Closes #

Changes proposed in this pull request:

* Added plate state page
* Added cherrytrack module
* Prettier'd some files

TODO:

Conditional filters based on whether the plate is a source or destination plate.
Work out filters needed for destination plate.
Update tests for cherrytrack plate endpoints.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,6 +1,7 @@
 SEQUENCESCAPE_BASE_URL=http://localhost:3000/api/v2
 LIGHTHOUSE_BASE_URL=http://localhost:5000
 LABWHERE_BASE_URL=http://localhost:3001/api
+CHERRYTRACK_BASE_URL=http://localhost:8000
```
Comment (harrietc52) [inline .env.example:4]: This requires deployment updates too

File: pages/plate_state.vue [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+      let plate = await lighthouseBiosero.getBioseroPlate(this.barcode, 'source')
+      if (plate.success) {
+        this.plate = plate
+      } else {
+        plate = await lighthouseBiosero.getBioseroPlate(this.barcode, 'destination')
+        if (plate.success) {
+          this.plate = plate
+        } else {
+          this.plate = { source: false, destination: false }
+          this.$refs.alert.show(`Could not find plate in Biosero with barcode: ${this.barcode}`, 'danger')
```
Comment (andrewsparkes) [inline pages/plate_state.vue:160]: Message sounds a bit odd. Maybe:
'Could not find a plate used on a Biosero system with the barcode: ${this.barcode}

File: test/layouts/default.spec.js [hunk 3]
```diff
@@ -31,7 +31,7 @@ describe('Index', () => {
 
   it('has a navbar', () => {
     expect(wrapper.findComponent({ ref: 'navbar' }).text()).toMatch(
-      /Lighthouse {2}Reports Box Buster Sentinel Sample Creation Sentinel Cherrypick Imports Print Labels Beckman Cherrypick Biosero Cherrypick UAT Actions/
+      /Lighthouse {2}Reports Box Buster Sentinel Sample Creation Sentinel Cherrypick Imports Print Labels Beckman Cherrypick Plate State Biosero Cherrypick UAT Actions/
```
Comment (andrewsparkes) [inline test/layouts/default.spec.js:34]: Biosero Plate State (if you do change it above)

**Review by andrewsparkes — APPROVED** [review]

File: pages/biosero_plate_state.vue [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+    return {
+      barcode: '',
+      lastPlateBarcode: '',
+      plate: { source: false, destination: false },
+      filter: 'source_barcode',
+      plateFields: [
+        { key: 'row', label: '', isRowHeader: true },
+        '1',
+        '2',
+        '3',
```
Comment (sdjmchattie) [inline pages/biosero_plate_state.vue:89]: Something doesn't feel right about this.  You've got a hash as the first entry in this array, but that feels like it shouldn't be part of the array itself as it looks like metadata for this whole array.  I think later this array is iterated over which confused me as to how this works.  Should this data be structured differently?  Something like:

```
plateFields: [
  {
    key: 'row',
    label: '',
    isRowHeader: true,
    values: ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12]
  }
]
```

Comment (BenTopping) [reply pages/biosero_plate_state.vue:89]: I agree it does look a little weird but the first hash is interpreted as a value with metadata by bootstrap so in itself it is a value. docs here which will explain it better than I do: 
https://bootstrap-vue.org/docs/components/table

Comment (sdjmchattie) [reply pages/biosero_plate_state.vue:89]: Thanks!  Today I learned something new.

**Review by sdjmchattie — APPROVED** [review]

**Pull request merged**